src/ui/states/state-error.svelte

```diff
@@ -1,49 +1,116 @@
+          );
         } else {
-          return "Already purchased";
+          return translator.translate(
+            LocalizationKeys.StateErrorErrorMessageAlreadyPurchased,
+          );
         }
       default:
-        return "Something went wrong";
+        return translator.translate(
+          LocalizationKeys.StateErrorErrorTitleOtherErrors,
+        );
+    }
+  }
+
+  function getTranslatedErrorMessage(): string | undefined {
+    const publicErrorCode = lastError.getErrorCode();
+    switch (lastError.errorCode) {
+      case PurchaseFlowErrorCode.UnknownError:
+        return translator.translate(
+          LocalizationKeys.StateErrorErrorMessageUnknownError,
+          { errorCode: publicErrorCode },
+        );
+      case PurchaseFlowErrorCode.ErrorSettingUpPurchase:
+        return translator.translate(
+          LocalizationKeys.StateErrorErrorMessageErrorSettingUpPurchase,
+          { errorCode: publicErrorCode },
+        );
+      case PurchaseFlowErrorCode.ErrorChargingPayment:
+        return translator.translate(
+          LocalizationKeys.StateErrorErrorMessageErrorChargingPayment,
+          { errorCode: publicErrorCode },
+        );
+      case PurchaseFlowErrorCode.NetworkError:
+        return translator.translate(
+          LocalizationKeys.StateErrorErrorMessageNetworkError,
+          { errorCode: publicErrorCode },
+        );
+      case PurchaseFlowErrorCode.StripeError:
+        // For stripe errors, we can display the stripe-provided error message.
+        return lastError.message;
+      case PurchaseFlowErrorCode.MissingEmailError:
+        return translator.translate(
+          LocalizationKeys.StateErrorErrorMessageMissingEmailError,
+          { errorCode: publicErrorCode },
+        );
+      case PurchaseFlowErrorCode.AlreadyPurchasedError:
+        if (productDetails?.productType === ProductType.Subscription) {
+          return translator.translate(
+            LocalizationKeys.StateErrorErrorMessageAlreadySubscribed,
+            { errorCode: publicErrorCode },
+          );
+        } else {
+          return translator.translate(
+            LocalizationKeys.StateErrorErrorMessageAlreadyPurchased,
+            { errorCode: publicErrorCode },
+          );
+        }
     }
   }
 </script>
 
 <MessageLayout
-  title={getErrorTitle()}
+  title={getTranslatedErrorTitle()}
   {brandingInfo}
   {onContinue}
   type="error"
-  closeButtonTitle="Try Again"
+  closeButtonTitle={translator.translate(
+    LocalizationKeys.StateErrorButtonTryAgain,
+  )}
 >
   <IconError slot="icon" />
 
-  {lastError.getPublicErrorMessage(productDetails)}
+  {getTranslatedErrorMessage()}
   {#if supportEmail}
-    <br>If this error persists, please contact <a href="mailto:{supportEmail}">{supportEmail}</a>.
+    <br />
+    <Localized labelId={LocalizationKeys.StateErrorIfErrorPersists}>
+      If this error persists, please contact
+    </Localized>
```

Comment:
We should remove the content here, right?
```suggestion
    <Localized labelId={LocalizationKeys.StateErrorIfErrorPersists} />
```